src/lambda/local/configureLocalLambda.ts

```diff
@@ -0,0 +1,245 @@
+    }
+
+    const handlersSymbol: vscode.DocumentSymbol | undefined = symbols.find(c => c.name === 'handlers')
+    if (!handlersSymbol || handlersSymbol.children.length < 1) {
+        const config: HandlersConfig = {
+            // Use jsonc-parser.parse instead of JSON.parse, as JSONC can handle comments. VS Code uses jsonc-parser
+            // under the hood to provide symbols for JSON documents, so this will keep us consistent with VS code.
+            ...parse(editor.document.getText()),
+            ...buildHandlersConfig(handler)
+        }
+        const configString = JSON.stringify(config, undefined, getTabSize(editor.options.tabSize))
+
+        return await editor.edit(editBuilder => editBuilder.replace(entireDocumentRange, configString))
+    }
+
+    const handlerSymbol: vscode.DocumentSymbol | undefined = handlersSymbol.children.find(c => c.name === handler)
+    if (!handlerSymbol || handlerSymbol.children.length < 1) {
+        // At this point we know that handlersSymbol has at least one child.
+        const lastChildEnd: vscode.Position = handlersSymbol.children.reduce(
+            (lastSoFar: vscode.Position, current: vscode.DocumentSymbol) =>
+                current.range.end.isAfter(lastSoFar) ? current.range.end : lastSoFar,
+            new vscode.Position(0, 0)
+        )
+
+        // For example (tabWidth = 4):
+        // [START],
+        //         "myHandler": {
+        //             event: {}
+        //         }
+        // [END]
+        const tabSize = getTabSize(editor.options.tabSize)
+        const baseIndentation: string = ' '.repeat(tabSize).repeat(2)
+        // We have already validated that handler contains only letters, numbers, hyphens, and underscores.
+        let snippet: string = `"${handler}": ${JSON.stringify(buildHandlerConfig(), undefined, tabSize)}`
+            .split(/\r?\n/).map(line => `${baseIndentation}${line}`).join(os.EOL)
+        snippet = `,${os.EOL}${snippet}${os.EOL}`
+
+        return await editor.edit(editBuilder => editBuilder.insert(lastChildEnd, snippet))
+    }
+
+    const eventSymbol: vscode.DocumentSymbol | undefined = handlerSymbol.children.find(c => c.name === 'event')
+    if (!eventSymbol) {
+        // At this point we know that handlerSymbol has at least one child.
+        const lastChildEnd: vscode.Position = handlerSymbol.children.reduce(
+            (lastSoFar: vscode.Position, current: vscode.DocumentSymbol) =>
+                current.range.end.isAfter(lastSoFar) ? current.range.end : lastSoFar,
+            new vscode.Position(0, 0)
+        )
+
+        const baseIndentation: string = ' '.repeat(getTabSize(editor.options.tabSize)).repeat(3)
+        const snippet: string = `,${os.EOL}${baseIndentation}"event": {}${os.EOL}`
+
+        return await editor.edit(editBuilder => editBuilder.insert(lastChildEnd, snippet))
+    }
+
+    return true
+}
+
+async function getFocusRange(editor: vscode.TextEditor, handler: string): Promise<vscode.Range> {
```

Comment:
This is called something general `getFocusRange` but is specific to finding the range of a handler's event configuration. For now, I suggest renaming it to something `event` centric, but I imagine this will be extended to search for additional fields in the future (like envvars)

Reply:
✔️